src/Components/Details.tsx

```diff
@@ -29,128 +39,357 @@ const excludedAttributes: Array<string> = [
+  return _result.join('\n');
+};
+
+const formatSharedMessage = (
+  general: Array<string[]>,
+  requestHeaders: Array<string[]>,
+  postData: string,
+  responseHeaders: Array<string[]>,
+  bodyResponse: string,
+): string => {
+  const _general = stringifyData(general);
+  const _requestHeaders = stringifyData(requestHeaders);
+  const _responseHeaders = stringifyData(responseHeaders);
+  const _report = ''.concat(
+    'GENERAL\n',
+    _general,
+    '\n',
+    'REQUEST HEADERS\n',
+    _requestHeaders,
+    '\n',
+    'REQUEST DATA\n',
+    postData,
+    '\n',
+    'RESPONSE HEADERS\n',
+    _responseHeaders,
+    '\n',
+    'RESPONSE BODY\n',
+    bodyResponse,
+  );
+  return _report;
+};
+
+export const Details: React.FC<IProps> = props => {
+  const theme = useContext(ThemeContext);
+  const [snackBarVisibility, setSnackBarVisibility] = useState<boolean>(false);
+  const [snackBarMessage, setSnackBarMessage] = useState<string>('');
+  let _content = null;
+  let _color: string = theme.reduxColor;
+  let _action: any = () => {};
+
+  const _onShareRequest = async (
+    general: Array<string[]>,
+    requestHeaders: Array<string[]>,
+    postData: string = '',
+    responseHeaders: Array<string[]>,
+    bodyResponse: string = '',
+  ): Promise<void> => {
+    try {
+      await Share.share({
+        message: formatSharedMessage(general, requestHeaders, postData, responseHeaders, bodyResponse),
+      });
+    } catch (error) {
+      Alert.alert('Error', error.message);
+    }
+  };
+
+  const _onShareReduxAction = async (item: ReduxAction): Promise<void> => {
+    const _type = item.action.type.toUpperCase();
+    const _payload = JSON.stringify(item.action.payload, null, 2);
+    try {
+      await Share.share({
+        message: `${_type}\n${_payload}`,
+      });
+    } catch (error) {
+      Alert.alert('Error', error.message);
+    }
+  };
+
+  const _copyToClipboard = (text: string): void => {
+    if (typeof text === 'string') Clipboard.setString(text);
+  };
+
+  const _copyClipbutton = (onPress: Function, text: string = '', toastMessage: string = '') => {
+    return (
+      <TouchableOpacity
+        testID="buttonCopyToClipboard"
+        style={[{ flexDirection: 'row', borderLeftWidth: 0, alignItems: 'center' }]}
+        onPress={() => {
+          onPress(text);
+          setSnackBarMessage(toastMessage);
+          setSnackBarVisibility(true);
+        }}
+      >
+        <MaterialCommunityIcons
+          name="clipboard-arrow-left-outline"
+          color={theme.primaryColor}
+          size={14}
+          testID="buttonCopyToClipBoard"
+        />
+        <Text style={[{ color: theme.primaryColor, marginLeft: 6 }]}>Copy</Text>
+      </TouchableOpacity>
+    );
+  };
+
+  const _renderItems = (listOfItems: Array<[string, any]>) => {
+    return listOfItems
+      .filter((item: Array<string>) => !excludedAttributes.includes(item[0]))
+      .filter((item: Array<string>) => item[1] && item[1].length > 0)
```

Comment:
we shoudl check item's length before `item[1]`

Reply:
Done